src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -653,6 +655,9 @@ public string DecryptToken(JsonWebToken jwtToken, TokenValidationParameters vali
             if (!decryptionSucceeded && keysAttempted.Length > 0)
                 throw LogHelper.LogExceptionMessage(new SecurityTokenDecryptionFailedException(LogHelper.FormatInvariant(TokenLogMessages.IDX10603, keysAttempted, exceptionStrings, jwtToken.EncodedToken)));
 
+            if (!decryptionSucceeded && algorithmNotSupportedByCryptoProvider)
```

Comment:
doing my final audit before merging this, noticed a logic error. this is claiming no algorithm is supported when the bool is checking one or more.